internal/awsdata/codecommit.go

```diff
@@ -0,0 +1,80 @@
+package awsdata
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/codecommit"
+	"github.com/manywho/awsinventory/internal/inventory"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// AssetTypeCodeCommitRepository is the value used in the AssetType field when fetching CodeCommit repositories
+	AssetTypeCodeCommitRepository string = "CodeCommit Repository"
+
+	// ServiceCodeCommit is the key for the CodeCommit service
+	ServiceCodeCommit string = "codecommit"
+)
+
+func (d *AWSData) loadCodeCommitRepositories(region string) {
+	defer d.wg.Done()
+
+	codecommitSvc := d.clients.GetCodeCommitClient(region)
+
+	log := d.log.WithFields(logrus.Fields{
+		"region":  region,
+		"service": ServiceCodeCommit,
+	})
+
+	log.Info("loading data")
+
+	var repositories []string
+	done := false
+	params := &codecommit.ListRepositoriesInput{}
+	for !done {
+		out, err := codecommitSvc.ListRepositories(params)
+
+		if err != nil {
+			d.results <- result{Err: err}
+			return
+		}
+
+		for _, repository := range out.Repositories {
+			repositories = append(repositories, aws.StringValue(repository.RepositoryName))
+		}
+
+		if out.NextToken == nil {
+			done = true
+		} else {
+			params.NextToken = out.NextToken
+		}
+	}
+
+	log.Info("processing data")
+
+	// TODO: API call can only handle 100 repository names at a time
+	out, err := codecommitSvc.BatchGetRepositories(&codecommit.BatchGetRepositoriesInput{
```

Comment:
I think we should check if the repo list is empty before this step and just return early. Currently if you have no code commit repos in your account, you get the following error:
```
ERRO[0000] RepositoryNamesRequiredException: repositoryNames cannot be empty
```

Reply:
This might be a problem in other loaders as well but I haven't tested yet

Reply:
Good catch! Fixed here and other places. Just went ahead and included those fixes in this PR.